### DISCUSSION TITLE
unexpected error:  'VegaFusionRuntime' object has no attribute 'set_connection'

### DISCUSSION THREAD

**Discussion by cboettig** [vega/vegafusion#548] [Q&A] [answered]
I have the vegafusion packages installed via conda:

```
vegafusion                2.0.0rc1        py312h63f49a4_0    conda-forge
vegafusion-jupyter        1.6.9              pyhd8ed1ab_0    conda-forge
vegafusion-python-embed   1.6.9           py312h63f49a4_2    conda-forge
```

I try to call `vf.runtime.set_connection` but get the error ` 'VegaFusionRuntime' object has no attribute 'set_connection'`

```python
import vegafusion as vf
import pandas as pd
import altair as alt

# Configure DuckDB connection
vf.runtime.set_connection("duckdb")
```

Produces the error trace:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 6
      4 import duckdb
      5 # Configure DuckDB connection
----> 6 vf.runtime.set_connection("duckdb")

AttributeError: 'VegaFusionRuntime' object has no attribute 'set_connection'
```

On Ubuntu 22.04 with python 3.12.7

**Comment by jonmmease** [comment, marked as answer]
Hi @cboettig, `set_connection` is being removed in VegaFusion 2, and you have the 2.0 release candidate installed (Unfortunately, conda is installing the release candidate version automatically).

DuckDB relations (`DuckDBPyRelation`) are now supported like inline datasets directly (See https://vegafusion.io/v2/posts/2024/2024-11-13_Release_2.0.0.html), so you don't need to `set_connection` and use `table://` URLs. Happy to answer any questions.

**Reply by cboettig** [reply]
awesome on all fronts.  (sorry I didn't notice about conda installing the release candidate by default!)

yup, things are working nicely as expected on 1.6.9, and the automatic support in 2.0 sounds fantastic.  really enjoying the altair/vega experience so far and hope to include this in my teaching!

**Reply by jonmmease** [reply]
Thanks for the kind words, and that's great to hear you're considering using this ecosystem for teaching.

Cross reference https://github.com/conda-forge/altair-feedstock/pull/58#issuecomment-2484439612 where @joelostblom ran into this same kind of issue with the release candidate.
